statistics/databuffer.go

```diff
@@ -0,0 +1,561 @@
+		results = append(results, metric)
+	}
+
+	if flag.SeventyFifthPercentile {
+		seventyfifthpercentile = d.PercentileNearestRank(75)
+		metric = createMetric(seventyfifthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("75%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyFirstPercentile {
+		ninetyfirstpercentile = d.PercentileNearestRank(91)
+		metric = createMetric(ninetyfirstpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("91%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyEighthPercentile {
+		ninetyeighthpercentile = d.PercentileNearestRank(98)
+		metric = createMetric(ninetyeighthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("98%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyFifthPercentile {
+		ninetyfifthpercentile = d.PercentileNearestRank(95)
+		metric = createMetric(ninetyfifthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("95%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.NinetyNinthPercentile {
+		ninetyninthpercentile = d.PercentileNearestRank(99)
+		metric = createMetric(ninetyninthpercentile, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("99%_ile")
+		results = append(results, metric)
+	}
+
+	if flag.Skewness {
+		skewness = d.Skewness(mean, standarddeviation)
+		metric = createMetric(skewness, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("skewness")
+		results = append(results, metric)
+	}
+
+	if flag.Kurtosis {
+		kurtosis = d.Kurtosis(mean, standarddeviation)
+		metric = createMetric(kurtosis, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("kurtosis")
+		results = append(results, metric)
+	}
+
+	if flag.Trimean {
+		trimean = d.Trimean(firstquartile, median, thirdquartile)
+		metric = createMetric(trimean, tags)
+		metric.Namespace = plugin.NewNamespace(nsPrefix...).AddStaticElements(ns...).AddStaticElement("trimean")
+		results = append(results, metric)
+	}
+	return results, err
+}
+
+// Returns count of the buffer
+func (d *dataBuffer) Count() int {
+	return len(d.data)
+}
+
+// Returns sum of all the data values in the buffer
+func (d *dataBuffer) Sum() (sum float64) {
+	for _, val := range d.data {
+		sum += val.value
+	}
+	return
+}
+
+// Returns the mean of the buffer
+func (d *dataBuffer) Mean(sum float64, count int) float64 {
+	return sum / float64(count)
+}
+
+// Returns the minimum value in the buffer
+func (d *dataBuffer) Minimum() float64 {
+	if len(d.data) > 0 {
+		return d.data[0].value
+	}
+
+	return math.NaN()
```

Comment:
as we cannot return math.NaN() due to json.Marshal() does not support it.  Let's check the length of buffer before call Minimum. Min always returns a float64. Please add a comment for it.